src/docker-osx-dev

```diff
@@ -829,10 +829,17 @@ function initial_sync {
   log_info "Performing initial sync of paths: ${paths_to_sync[@]}"
 
   local dirs_to_create=()
+  local parent_dirs=()
+  local base_names=()
   local path=""
+  local parent_dir=""
```

Comment:
Within the for loop, `parent_dir` and `base_name` (not `parent_dirs` or `base_names`) are written just once, so `readonly` would work. As for bash scoping, I'm not sure (easy enough to check), but do you need two for loops? Any reason not to do the tar in first loop?